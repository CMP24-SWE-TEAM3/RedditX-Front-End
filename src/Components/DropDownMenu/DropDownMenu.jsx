import Dropdown from "react-bootstrap/Dropdown";
import React from "react";
import DropDownItem from "Components/DropDownItem/DropDownItem";
import {DropDownMenuStyled} from "./DropDownMenu.styled";

const DropDownMenu = () => {
    return (
        <DropDownMenuStyled>
            <Dropdown.Menu id={'drop-down-menu-home'} show={true}>
                <DropDownItem/>
            </Dropdown.Menu>
        </DropDownMenuStyled>
    );
};

export default DropDownMenu;
