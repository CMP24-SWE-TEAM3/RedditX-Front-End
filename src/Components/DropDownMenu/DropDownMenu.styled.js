import styled from "styled-components";

export const DropDownMenuStyled = styled.div`
  position: absolute;
  .dropdown-menu {
    background-color: ${({theme}) => theme.background.primary};
    border: 1px solid white;
    border-radius: 0 0 4px 4px;
    box-sizing: border-box;
    margin-top: -1px;
    height: 482px;
    overflow: hidden scroll;
    position: sticky;
    display: flex;
    -ms-flex-direction: column;
    flex-direction: column;
    left: 0;
    bottom: 0;
    right: 0;
    top: 48px;
    width: 270px;
    
    ul {
      padding-top: 10px;
    }
  }

  .form-control {
    background-color: ${({theme}) => theme.background.post_background};
    border: 1px solid ${({theme}) => theme.color.secondary};
    color: ${({theme}) => theme.color.primary};
    height: 30px;
    margin: 16px 16px 0;
    outline: none;
    padding: 0 6px;
    box-sizing: border-box;
    border-radius: 0;
    width: calc(100% - 32px);
  }

  .form-control:focus {
    border-color: ${({theme}) => theme.color.secondary} !important;
    box-shadow: none;
  }
`