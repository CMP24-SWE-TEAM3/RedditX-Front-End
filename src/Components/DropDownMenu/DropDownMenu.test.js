import {shallow} from 'enzyme';

// import Components
import DropDownMenu from "./DropDownMenu";

describe("Drop Down Menu Component", () => {
    it("this is a test for Drop Down Menu", () => {
        expect(shallow(<DropDownMenu/>)).toMatchSnapshot();

    });
});