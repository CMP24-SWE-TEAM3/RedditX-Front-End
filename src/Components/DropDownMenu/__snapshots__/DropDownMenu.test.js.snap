// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drop Down Menu Component this is a test for Drop Down Menu 1`] = `ShallowWrapper {}`;
